crates/miden-objects/src/transaction/tx_args.rs

```diff
@@ -104,14 +103,14 @@ impl TransactionArgs {
     /// - inputs_key |-> inputs, where inputs_key is computed by taking note inputs commitment and
     ///   adding ONE to its most significant element.
     /// - script_root |-> script.
-    pub fn add_expected_output_note<T: Deref<Target = NoteDetails>>(&mut self, note: &T) {
-        let recipient = note.recipient();
-        let inputs = note.inputs();
-        let script = note.script();
+    pub fn add_expected_output_note<T: AsRef<NoteRecipient>>(&mut self, recipient: T) {
```

Comment:
I think the we should update the doc comments a bit. But also, does the naming still make sense here? We are no longer adding "notes" - so, maybe the name of the method should be something else.

Reply:
I debated this but eventually decided not to. In the context where it is used it still makes sense to "add expected notes"(maybe not in the client changes), and `T` can still be a note type. 
But not opposed to do it at all, although I'm not sure what the best alternative would be. `add_output_note_recipient_details()`?

Reply:
Yeah, I'm on the fence about this too. As for the function name `add_output_note_recipient_details()` could work, or we could even do `add_output_note_recipients()`.